src/context/AuthContext.tsx

```diff
@@ -0,0 +1,74 @@
+import { ReactNode } from "react";
+import { createContext, useEffect, useReducer } from "react";
+import { auth } from "firebase/config";
+import { onAuthStateChanged } from "firebase/auth";
+import { UserDataType } from "types/UserDataType";
+
+type AuthContextType = {
+	user?: UserDataType | null;
+	dispatch: React.Dispatch<AuthContextAction>;
+	isUserLoading?: boolean;
+} | null;
+
+interface AuthContextAction {
+	type: "LOGIN" | "LOGOUT" | "LOGGED";
+	payload?: UserDataType | unknown;
+	isUserLoading?: boolean;
+}
+
+interface AuthContextState {
+	user: UserDataType | null;
+}
+
+type AuthContextProviderProps = {
+	children: ReactNode;
+};
+
+export const AuthContext = createContext<AuthContextType>(null);
+
+const reducer = (state: AuthContextState, action: AuthContextAction) => {
+	const { type, payload, isUserLoading } = action;
+
+	switch (type) {
+		case "LOGIN":
+			return {
+				...state,
+				user: payload as UserDataType,
+			};
+		case "LOGOUT":
+			return { ...state, user: null };
```

Comment:
**Różnica pomiędzy undefined i null**

Różnica jest taka, że `null` reprezentuje pustą wartość, a `undefined` informuje, ze czegoś nie ma.
Lepsze byłoby gdyby klucz user nie był "pustą wartością", ale żeby go po prostu nie było.

Jak to zrobisz to rozwiąże Ci się problem z linii 36.